components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/SubToolbar.kt

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.toolbar
+
+import android.content.Context
+import android.transition.TransitionManager
+import android.view.View
+import mozilla.components.browser.toolbar.display.DisplayToolbar
+import mozilla.components.browser.toolbar.edit.EditToolbar
+import android.view.ViewGroup
+import mozilla.components.concept.toolbar.Toolbar
+
+/**
+ * Abstract base class for shared code between [DisplayToolbar] and [EditToolbar].
+ */
+internal abstract class SubToolbar(
```

Comment:
nit: This name confused me a bit i.e. why is it a `Sub` toolbar if `Display` and `Edit` toolbar extend it? Maybe just `AbstractToolbar`?

Reply:
I took `SubToolbar` since it's only extended by the two "sub" components and not the toolbar itself.

```
            +----------------+
            | BrowserToolbar |
            +--------+-------+
                     +
             +-------+-------+
             |               |
   +---------v------+ +-------v--------+
   | DisplayToolbar | |   EditToolbar  |
   +----------------+ +----------------+
```

I feel like `AbstractToolbar` sounds like it's used by the `BrowserToolbar` which implements `Toolbar`. Confusing :D

Reply:
Actually nothing in `SubToolbar` requires any state or requires this to be an abstract class. I think I'm going to move the things in it to be just functions.. 🤔

Reply:
Ah ok! I didn't get that :). The new version looks great though!